### ISSUE TITLE
imports folder missing from tutorial

### ISSUE THREAD

**Issue by GrilloPress** [meteor/tutorials#53]
Following the https://www.meteor.com/tutorials/blaze/creating-an-app tutorial.

In the first section you run `meteor create` and you have a folder structure of

```
client/main.js        # a JavaScript entry point loaded on the client
client/main.html      # an HTML file that defines view templates
client/main.css       # a CSS file to define your app's styles
server/main.js        # a JavaScript entry point loaded on the server
package.json          # a control file for installing NPM packages
.meteor               # internal Meteor files
.gitignore            # a control file for git
```

However, in the second section of the tutorial https://www.meteor.com/tutorials/blaze/templates we are told to add some files within an `import` folder.

Namely "Then we create some new files in the imports/ directory:"

This doesn't exist. 

I see from http://guide.meteor.com/structure.html that this import folder should wrap everything. Is this something we should be told to wrap things within? At this second, the guide just assumes we have an import directory, which we don't have.

**Comment by stubailo** [comment]
I guess we assumed that people would understand that means "create an imports directory, then put files there". But we should improve the text to say that, or have `meteor create` generate an empty `imports` directory.

**Comment by GrilloPress** [comment]
You are right most people will create a directory somewhere but think it would be better with explicit direction.

Worth just adding a step, as you say, to indicate that the reader should create a directory called `imports` or having the generator do that already (depending on how established the imports directory structure will be)

**Comment by kepoly** [comment]
I was also a bit confused with this @stubailo if you were to auto generate like you mentioned or just leave a little sticky above 3.1 that says "if you haven't already created the imports folder do so now" that would be great for future learners!

**Comment by wbeeftink** [comment]
I also started checking step 1 to see if I had missed anything. The idea of adding the text like @stubailo  mentions is nice.

**Comment by billybatsaikhan** [comment]
So did you guys create the imports folder inside the simple-todos folder? @stubailo @GrilloPress @wbeeftink @kepoly @benjamn

**Comment by GrilloPress** [comment]
@Batsaikhan-bilguun short answer: yes.

Just create an imports directory alongside your client and server directories and then follow the tutorial, placing any imports folders/code into that directory.

Your project should have the top-level structure similar to this:

``` shell
imports/              # ADD THIS DIRECTORY MANUALLY
client/               # All your client  code inside here
server/               # All your server code inside here
package.json          # a control file for installing NPM packages
.meteor               # internal Meteor files
.gitignore            # a control file for git
```

**Comment by kahlilashanti** [comment]
I've created it manually in the simple-todos directory and I'm still getting this error message: install.js:78Uncaught Error: Cannot find module '../imports/ui/body.js'

I'm on a new MacBook Air using Chrome.

How can I create it manually using 'meteor create'?

**Comment by GrilloPress** [comment]
@kahlilashanti a bit tough to help troubleshoot your issue without looking at the code, but your app cannot find the body.js file within your imports/ directory.

Do you have an imports directory with an UI directory inside it? Does your project folder contain something like this:

``` bash
imports/
       ui/
         body.js
         body.html
```

Have you created that file as per the tutorial code?

Unless I'm mistaken, the `meteor create` command just creates the initial meteor app structure and as of now, it doesn't create the `imports` directory for you.

**Comment by kahlilashanti** [comment]
That's correct - my code looks like the example you provided.

On Sun, May 15, 2016 at 6:10 AM, Andrew Duckworth notifications@github.com
wrote:

> @kahlilashanti https://github.com/kahlilashanti a bit tough to help
> troubleshoot your issue without looking at the code, but your app cannot
> find the body.js file within your imports/ directory.
> 
> Do you have an imports directory with an UI directory inside it? Does your
> code look something like this:
> 
> imports/
>        ui/
>          body.js
> 
> Have you created that file as per the tutorial code?
> 
> Unless I'm mistaken, the meteor create command just creates the initial
> meteor app structure and as of now, it doesn't create the imports
> directory for you.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/meteor/tutorials/issues/53#issuecomment-219284933

## 

Kahlil Ashanti
www.kahlilashanti.com

**Comment by gdoteof** [comment]
I had a similar issue and realized I had my `App.jsx` named `Apps.jsx`.  Anyone who finds themselves here should be relatively certain they have a typo/misnamed file/accidental file structure.  If you really can't find it and insist on posting here, I would recommend including the file doing the import, as well as a complete dump of your folder structure.

ie:

```
cat THE_IMPORTING_FILE
find /path/to/project/folder
```

**Comment by thefuturewasnow** [comment]
Thanks GrilloPress!

**Comment by dirkdunn** [comment]
Thank god I found this, the way it read made it seem like there should have been an import directory already generated! Definitely needs to be added to the tutorial, NOT intuitive.

**Comment by darkmakukudo** [comment]
The documentation should be updated tell me how so I can contribute

**Comment by Marcelovv2** [comment]
Thanks, this is needed specially for beginners like me.

**Comment by lazaridis-com** [comment]
> or have meteor create generate an empty imports directory.

@abernix,  @stubailo, the create command should create the `import` directory.

**Comment by endymion** [comment]
I agree.  It should go like this: Document this immediately because this is confusing a lot of people.  People on my team tripped on this today too, and it looks like lots of other people do.
And then make the create command create the directory automatically and remove the apology from the documentation.

**Comment by lazaridis-com** [comment]
@hwillson @abernix - please just merge #86, this would solve this issue.

**Comment by abernix** [comment]
Solved with a combination of #86 and 41d7a4f57f68ef8e18eda45127f937065714dd4b.  We still aren't going to automatically create the `imports` directory with the basic `meteor create`, but we now explain its significance and the exercise of adding it in the tutorial should make it clear what purpose it serves – it is a magic directory, after all!

If you want the directory automatically created, you can consider using the `--full` argument to `meteor create`:

    meteor create app-name --full

...which will create a much more structured application, in line with what is recommended in the [Meteor Guide](https://guide.meteor.com/).

Sincere apologies to everyone who has ever encountered this, and even more so to the ones who actually took the time to report it.

This isn't live yet, but will be by the end of the day (I'll be pushing a bunch of tutorial updates together)
